File: 프로그래머스/src/프로그래머스_02262022/같은_숫자는_싫어.cpp
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr)
{
    vector<int> answer;
    //연속만 아니면 됌 -> peek
    for (auto a : arr)
    {
        if (!answer.empty() & answer.back() == a)
            continue;
        else
            answer.push_back(a);
    }

    return answer;
}

File: 2026/Level1/같은 숫자는 싫어.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12906?language=python3
def solution(arr):
    # Base Case
    if len(arr) == 0 :
        return arr
    
    prev = arr[0]
    answer = [prev]
    for x in arr :
        if prev != x  :
            answer.append(x)
        prev = x
    
    return answer